src/view-backend-exportable-private.cpp

```diff
@@ -111,20 +111,36 @@ void ViewBackend::dispatchFrameCallbacks()
     }
     clearFrameCallbacks();
 
-    wl_client_flush(m_client);
+    if (m_client)
+        wl_client_flush(m_client);
     wpe_view_backend_dispatch_frame_displayed(m_backend);
 }
 
 void ViewBackend::releaseBuffer(struct wl_resource* buffer_resource)
 {
     wl_buffer_send_release(buffer_resource);
-    wl_client_flush(m_client);
+    if (m_client)
+        wl_client_flush(m_client);
 }
 
 void ViewBackend::registerSurface(uint32_t surfaceId)
 {
     m_surfaceId = surfaceId;
     m_client = WS::Instance::singleton().registerViewBackend(m_surfaceId, *this);
+ 
+    struct wl_client_destroy_listener *listener = new wl_client_destroy_listener {this, };
+    listener->destroyClientListener.notify = (wl_notify_func_t) [](struct wl_listener* listener, void* data)
+    {
+        struct wl_client_destroy_listener *container;
+        container = wl_container_of(listener, container, destroyClientListener);
+
+        struct wl_client* client = (struct wl_client*) data;
+        g_debug("ViewBackend <%p>: wl_client <%p> destroy notification received for fd %d", container->backend, data, wl_client_get_fd(client));
```

Comment:
I suppose we don't want to keep this debugging code in the final version of the patch, right?

Reply:
Yes, makes senses.

Reply:
I've cherry-picked the cleanups suggested by @zdobersek. Also, I0ve verified the `unregisterViewBackend()` is called when the  WebView is destroyed (ex: `g_object_unref()`) so It looks like the right place to put the call for the  `wl_client_destroy()`.